datafusion/physical-expr/src/expressions/any.rs

```diff
@@ -0,0 +1,570 @@
+
+        let negated = match self.op {
+            Operator::Eq => false,
+            Operator::NotEq => true,
+            op => {
+                return Err(DataFusionError::NotImplemented(format!(
+                    "Operator for ANY expression, actual: {:?}",
+                    op
+                )));
+            }
+        };
+
+        match value.data_type() {
+            DataType::Float16 => {
+                make_primitive!(value, as_list, negated, Float16Array, list_from_scalar)
+            }
+            DataType::Float32 => {
+                make_primitive!(value, as_list, negated, Float32Array, list_from_scalar)
+            }
+            DataType::Float64 => {
+                make_primitive!(value, as_list, negated, Float64Array, list_from_scalar)
+            }
+            DataType::Int8 => {
+                make_primitive!(value, as_list, negated, Int8Array, list_from_scalar)
+            }
+            DataType::Int16 => {
+                make_primitive!(value, as_list, negated, Int16Array, list_from_scalar)
+            }
+            DataType::Int32 => {
+                make_primitive!(value, as_list, negated, Int32Array, list_from_scalar)
+            }
+            DataType::Int64 => {
+                make_primitive!(value, as_list, negated, Int64Array, list_from_scalar)
+            }
+            DataType::UInt8 => {
+                make_primitive!(value, as_list, negated, UInt8Array, list_from_scalar)
+            }
+            DataType::UInt16 => {
+                make_primitive!(value, as_list, negated, UInt16Array, list_from_scalar)
+            }
+            DataType::UInt32 => {
+                make_primitive!(value, as_list, negated, UInt32Array, list_from_scalar)
+            }
+            DataType::UInt64 => {
+                make_primitive!(value, as_list, negated, UInt64Array, list_from_scalar)
+            }
+            DataType::Boolean => {
+                self.compare_bool(value, as_list, negated, list_from_scalar)
+            }
+            DataType::Utf8 => {
+                self.compare_utf8::<i32>(value, as_list, negated, list_from_scalar)
+            }
+            DataType::LargeUtf8 => {
+                self.compare_utf8::<i64>(value, as_list, negated, list_from_scalar)
+            }
+            datatype => Result::Err(DataFusionError::NotImplemented(format!(
+                "AnyExpr does not support datatype {:?}.",
+                datatype
+            ))),
+        }
+    }
+}
+
+/// return two physical expressions that are optionally coerced to a
+/// common type that the binary operator supports.
+fn any_cast(
+    value: Arc<dyn PhysicalExpr>,
+    _op: &Operator,
+    list: Arc<dyn PhysicalExpr>,
+    input_schema: &Schema,
+) -> Result<(Arc<dyn PhysicalExpr>, Arc<dyn PhysicalExpr>)> {
+    let tmp = list.data_type(input_schema)?;
+    let list_type = match &tmp {
+        DataType::List(f) => f.data_type(),
+        _ => panic!("wtf"),
```

Comment:
We should return `Err` here, with a slightly more meaningful error message ;-)

Reply:
Oops! My bad :D Fixed.